libcontainer/configs/config.go

```diff
@@ -31,10 +31,12 @@ type IDMap struct {
 // for syscalls. Additional architectures can be added by specifying them in
 // Architectures.
 type Seccomp struct {
-	DefaultAction   Action     `json:"default_action"`
-	Architectures   []string   `json:"architectures"`
-	Syscalls        []*Syscall `json:"syscalls"`
-	DefaultErrnoRet *uint      `json:"default_errno_ret"`
+	DefaultAction    Action     `json:"default_action"`
+	Architectures    []string   `json:"architectures"`
+	Syscalls         []*Syscall `json:"syscalls"`
+	DefaultErrnoRet  *uint      `json:"default_errno_ret"`
+	ListenerPath     string     `json:"listenerPath,omitempty"`
+	ListenerMetadata string     `json:"listenerMetadata,omitempty"`
```

Comment:
Seems mixing up snake_case and camelCase

Reply:
Ouch, thanks! Will fix

Reply:
I’ve changed it to keep the style currently in master (came case for struct field names and underscore for json tags), thanks!

I think we probably can't change this, as it is in libcontainer and might break users. But it wasn't obvious why it was working for me, so looking at for example the DefaultErrnoRet field seems to be parsed correctly today on master because from [json unmarshall doc][link]:

> Unmarshal matches incoming object keys to the keys used by Marshal (either the struct field name or its tag), preferring an exact match but also accepting a case-insensitive match”. So, it all works because the struct field name is as expected (and match is case insensitive if needed, and we need that as well as the struct field name starts with a capital letter but not the json field name), making basically the json tags quite pointless.

And then, I guess it works when the json has `"defaultErrnoRet": 6` because the struct field name is DefaultErrnoRet and matches case-insensitive (defaultErrnoRet), and the json tag is mostly unused (as probably most users use camel case instead of underscores, as that is how the field is defined in the runtime spec).

Did the same (so basically camel case and underscore works) for the ListenerPath and ListenerMetadata fields.

Thanks!

[link]: https://pkg.go.dev/encoding/json#Unmarshal